web/assets/js/actions.ts

```diff
@@ -102,10 +105,13 @@ class Actions {
   public async setFaucetToken(token: string) {
     this.faucetToken = token;
 
-    localStorage.setItem(defaultFaucetTokenKey, token);
-
     // Attempt to fund the account
-    await this.fundAccount();
+    try {
+      await this.fundAccount();
+      localStorage.setItem(defaultFaucetTokenKey, token);
```

Comment:
@zivkovicmilos If the method can return an error because the user is funded already, do we have a good way to know the error comes from an invalid token?

Reply:
Yes, I added a method that will return a typed err:
https://github.com/gnolang/gnochess/blob/8915db3341953d4216e65cfe68b62482dc2890cb/web/assets/js/utils/errors.ts#L7-L15

Reply:
`fundAccount` will throw one of these, you can check it on your end

Reply:
You are the best! Thank you!!